### PR TITLE
Add a is_shipping_line flag to view_cart xhr response

### PR DESCRIPTION
**Pull request by samakshjain**
This change is [Reviewable](https://reviewable.io/reviews/fulfilio/shop/102)

### CODE REVIEW

File: shop/cart/views.py [hunk 1]
```diff
@@ -33,9 +33,9 @@ def view_cart():
...
                     'unit_price': l.unit_price.format(current_locale),
                     'amount': l.amount.format(current_locale),
-                    'url': l.product.listing.get_absolute_url(),
+                    'url': l.product and l.product.listing and l.product.listing.get_absolute_url(),
                     'image': l.product.image,
-                } for l in cart.sale.lines],
+                } for l in cart.sale.lines if not l.shipment_cost],
                 'empty': len(cart.sale.lines) < 1,
                 'total_amount': cart.sale.total_amount.format(current_locale),
                 'tax_amount': cart.sale.tax_amount.format(current_locale),
```
Comment (prakashpp) [inline shop/cart/views.py]: also return shipment cost

**Comment by sharoonthomas** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[shop/cart/views.py, line 38 at r2](https://reviewable.io:443/reviews/fulfilio/shop/102#-KV0Fhy_65S_A1daDJIF:-KV0Fhy_65S_A1daDJIG:b-buhguo) ([raw file](https://github.com/fulfilio/shop/blob/db7255e1b8903dc0f5f25a718af183515b2836c4/shop/cart/views.py#L38)):*
> ```Python
>                     'url': l.product and l.product.listing and l.product.listing.get_absolute_url(),
>                     'image': l.product.image,
>                 } for l in cart.sale.lines if not l.shipment_cost],
> ```

I don't think this is the right approach.

All lines should be returned. The client should decide whether to render this or not.

In addition, there is a boolean flag called "is_shipping_line" which should be sent IMO

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fulfilio/shop/102)*

**Comment by samakshjain** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[shop/cart/views.py, line 41 at r1](https://reviewable.io:443/reviews/fulfilio/shop/102#-KUwUv0sw4tl06DjB_dw-r1-41:-KV0OcPaZpwaTAsUMRkk:b-896fix) ([raw file](https://github.com/fulfilio/shop/blob/ebedc0e50dd033367a07cd2bd9c126d0903d9af9/shop/cart/views.py#L41)):*
<details><summary><i>Previously, prakashpp (Prakash Pandey) wrote…</i></summary>
> also return shipment cost
</details>
Done.

---

*[shop/cart/views.py, line 38 at r2](https://reviewable.io:443/reviews/fulfilio/shop/102#-KV0Fhy_65S_A1daDJIF:-KV0OALQQ8MIrbPH1Uv2:b-8e5kr4) ([raw file](https://github.com/fulfilio/shop/blob/db7255e1b8903dc0f5f25a718af183515b2836c4/shop/cart/views.py#L38)):*
<details><summary><i>Previously, sharoonthomas (Sharoon Thomas) wrote…</i></summary>
> I don't think this is the right approach.
> 
> All lines should be returned. The client should decide whether to render this or not.
> 
> In addition, there is a boolean flag called "is_shipping_line" which should be sent IMO
</details>
I couldn't find a `is_shipping_line` boolean field, there was a field close to the meaning named `is_channel_shipping`. Should I use that instead?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fulfilio/shop/102)*

**Comment by samakshjain** [comment]
This now depends on: https://github.com/fulfilio/lcf-shop/pull/113

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fulfilio/shop/102#-:-KV0P2hJqm9_ZcKNaYsc:b2p78mk)*

**Pull request merged**